src/FormParserFactory.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace React\Http;
+
+use React\Http\Parser\FormUrlencoded;
+use React\Http\Parser\Multipart;
+
+class FormParserFactory
+{
+    public static function create(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        if (!array_key_exists('content-type', $headers)) {
```

Comment:
Haven't though out what to do what to return when we can't determine which parser to use. Not a fan of returning false and have mixed return types. Thinking of either returning a special class that does nothing or a way to check out front if we can return anything useful. /cc @clue